Assets/Scripts/Hierarchical/HierarchicalBase.cs

```diff
@@ -59,27 +63,38 @@ public void RemovePursuer(IHierarchical pursuer) {
     _pursuers.Remove(pursuer);
   }
 
-  protected virtual Vector3 GetPosition() {
-    return GetMean(s => s.Position);
-  }
-
-  protected virtual Vector3 GetVelocity() {
-    return GetMean(s => s.Velocity);
-  }
-
-  protected virtual Vector3 GetAcceleration() {
-    return GetMean(s => s.Acceleration);
+  public bool IsEscapingPursuers() {
+    return Pursuers.All(pursuer => {
+      // A hierarchical object is considered escaping a pursuer if the closing velocity is
+      // non-positive or if the hierarchical object will reach its target before the pursuer reaches
+      // the hierarchical object.
+      Vector3 relativePosition = pursuer.Position - Position;
+      Vector3 relativeVelocity = pursuer.Velocity - Velocity;
+      float rangeRate = Vector3.Dot(relativeVelocity, relativePosition.normalized);
+      float closingVelocity = -rangeRate;
+      if (closingVelocity <= 0) {
+        return true;
+      }
+      float pursuerDistance = relativePosition.magnitude;
+      float targetDistance = (Target.Position - Position).magnitude;
+      float timeToTarget = targetDistance / Speed;
+      float pursuerTimeToIntercept = pursuerDistance / pursuer.Speed;
+      return pursuerDistance > targetDistance && timeToTarget < pursuerTimeToIntercept;
+    });
```

Comment:
If a hierarchical object doesn't have a target assigned (e.g., a newly created launcher before clustering), calling `IsEscapingPursuers()` will throw an NRE.

If Target is null should we return false?

Reply:
Yes, I'll add that.  `IsEscapingPursuers` is only called by threats, which always have a target.